src/geofabrics/processor.py

```diff
@@ -51,8 +53,17 @@ def run(self):
         window_size = 0
         idw_power = 2
         
-        ### Load in LiDAR using PDAL
-        catchment_lidar = lidar.CatchmentLidar(self.instructions['instructions']['data_paths']['lidars'][0], self.catchment_geometry)
+        ### Ensure LiDAR data is downloaded within the catchment region from OpenTopography
+        if 'local_cache' in self.instructions['instructions']['data_paths']:  # download from OpenTopography - then get the local file path
+            self.lidar_fetcher = lidar_fetch.OpenTopography(self.catchment_geometry, self.instructions['instructions']['data_paths']['local_cache'], verbose = True)
+            self.lidar_fetcher.run()
+            
+            lidar_file_paths = list(pathlib.Path(self.lidar_fetcher.cache_path / self.lidar_fetcher.dataset_prefixes[0]).glob('*.laz'))
```

Comment:
What is the reason for the `/ self.lidar_fetcher.dataset_prefixes[0]`? A comment would help, but maybe its just my poor understanding of the file structure

Reply:
Oh - yes - basically I haven't implemented support for merging multiple lidar datasets yet. I also haven't added support (its a separate issue) for considering multiple lidar tiles. I've added two variables at the top of the run method and a comment that hopefully makes this a lot clearer. It really wasn't as you found it!